scripting/src/main/java/org/infinispan/scripting/impl/DistributedRunner.java

```diff
@@ -34,13 +35,12 @@ private DistributedRunner() {
       DefaultExecutorService des = new DefaultExecutorService(masterCacheNode);
       try {
          List<Future<T>> tasks = des.submitEverywhere(new DistributedScript<T>(metadata));
-         List<NotifyingFuture<T>> notifyingTasks = new ArrayList<>(tasks.size());
-         for (Future<T> task : tasks) {
-               notifyingTasks.add((NotifyingFuture<T>) task);
+         List<CompletableFuture<T>> all = new ArrayList<>(tasks.size());
```

Comment:
asCompletableFuture needs NotifyingFutures because it uses the attachListener method